File: spring-boot-backend/src/main/java/com/springbootbackend/model/CartItem.java

package com.springbootbackend.model;


import jakarta.persistence.*;
        import lombok.*;

@Data
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name= "cart_item")
public class CartItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long cartItemId;

    @ManyToOne
    @JoinColumn(name = "cart_id",nullable = false)
    private Cart cart;

    @ManyToOne
    @JoinColumn(name = "product_id",nullable = false)
    private Product product;

    private Integer quantity;


}

File: flask-backend/models/cart_item.py
from extensions import db


class CartItem(db.Model):
    cartItemId = db.Column(db.Integer, primary_key=True)
    cartId = db.Column(db.Integer, db.ForeignKey("cart.cartId"), nullable=False)
    productId = db.Column(
        db.Integer, db.ForeignKey("products.productId"), nullable=False
    )
    quantity = db.Column(db.Integer, default=1)

    product = db.relationship("Product", lazy="joined")
